clippy_lints/src/needless_pass_by_ref_mut.rs

```diff
@@ -241,8 +241,10 @@ impl<'tcx> LateLintPass<'tcx> for NeedlessPassByRefMut<'tcx> {
             .iter()
             .filter(|(def_id, _)| !self.used_fn_def_ids.contains(def_id))
         {
-            let show_semver_warning =
-                self.avoid_breaking_exported_api && cx.effective_visibilities.is_exported(*fn_def_id);
+            let is_exported = cx.effective_visibilities.is_exported(*fn_def_id);
+            if self.avoid_breaking_exported_api && is_exported {
+                continue;
+            }
```

Comment:
Remove this condition and I think it'll fix the regression.

Reply:
But this condition is what this PR is about :sweat_smile:

Reply:
No, it's about not emitting the error when `self.avoid_breaking_exported_api` is false. :3

Reply:
Yeah, so if we want to avoid breaking exported API and the function is exported, we want to skip the lint emission -> `continue;`. What am I missing?

Reply:
We collect callbacks to check where they're called to see if the arguments of a callback actually need to be a `&mut`. (At least that's what I understood from the test failure)